docs/guides/mlp_tutorials/llm-inference.md

```diff
@@ -0,0 +1,254 @@
+
+Now that we've setup the Dockerfile, we can go ahead and pass it to [Podman](https://podman.io/) to build a container. Podman is a tool that enables us to fetch, manipulate, and interact with containers on the cluster. For more information, please see the [Container Engine][ref-container-engine] page. To use Podman, we first need to configure some storage locations for it. This step is straightforward, just make the file `$HOME/.config/containers/storage.conf` (or `$XDG_CONFIG_HOME/containers/storage.conf` if `XDG_CONFIG_HOME` is set):
+
+```
+[storage]
+  driver = "overlay"
+  runroot = "/dev/shm/$USER/runroot"
+  graphroot = "/dev/shm/$USER/root"
+
+[storage.options.overlay]
+  mount_program = "/usr/bin/fuse-overlayfs-1.13"
+```
+
+
+To build a container with Podman, we need to request a shell on a compute node from [SLURM][ref-slurm], pass the Dockerfile to Podman, and finally import the freshly built container using enroot. SLURM is a workload manager which distributes workloads on the cluster. Through SLURM, many people can use the supercomputer at the same time without interfering with one another in any way:
+
+```
+[cluster][user@cluster-ln001 pytorch-24.01-py3-venv]$ srun -A <ACCOUNT> --pty bash
+[cluster][user@nid001234 pytorch-24.01-py3-venv]$ podman build -t pytorch:24.01-py3-venv .
+# ... lots of output here ...
+[cluster][user@nid001234 pytorch-24.01-py3-venv]$ enroot import -x mount -o pytorch-24.01-py3-venv.sqsh podman://pytorch:24.01-py3-venv
+# ... more output here ...
+```
+
+where you should replace `<ACCOOUNT>` with your project account ID. At this point, you can exit the SLURM allocation by typing `exit`. You should be able to see a new squashfile next to your Dockerfile:
+
+```
+[cluster][user@cluster-ln001 pytorch-24.01-py3-venv]$ ls
+Dockerfile  pytorch-24.01-py3-ven.sqsh
+```
+
+This squashfile is essentially a compressed container image, which can be run directly by the container engine. We will use our freshly-built container `pytorch-24.01-py3-venv.sqsh` in the following steps to run a PyTorch script that loads the Google Gemma-7B model and performs some inference with it.
+
+### Set up an EDF
+
+We need to set up an EDF (Environment Definition File) which tells the Container Engine what container to load, where to mount it, and what plugins to load. Use your favorite text editor to create a file `~/.edf/gemma-pytorch.toml` for the container engine. The EDF should look like this:
+
+```
+image = "/capstor/scratch/cscs/<USER>/pytorch-24.01-py3-venv/pytorch-24.01-py3-venv.sqsh"
+
+mounts = ["/capstor", "/users"]
+
+writable = true
+
+[annotations]
+com.hooks.aws_ofi_nccl.enabled = "true"
+com.hooks.aws_ofi_nccl.variant = "cuda12"
+
+[env]
+FI_CXI_DISABLE_HOST_REGISTER = "1"
+FI_MR_CACHE_MONITOR = "userfaultfd"
+NCCL_DEBUG = "INFO"
```

Comment:
These environment variables need not be set manually anymore. Instead maybe refer to [ref-communication-nccl] for more details.